ee/clickhouse/views/experiments.py

```diff
@@ -353,3 +356,89 @@ def requires_flag_implementation(self, request: Request, *args: Any, **kwargs: A
         warning = requires_flag_warning(filter, self.team)
 
         return Response({"result": warning})
+
+    @action(methods=["POST"], detail=True)
+    def create_exposure_cohort_for_experiment(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        experiment = self.get_object()
+        flag = getattr(experiment, "feature_flag", None)
+        if not flag:
+            raise ValidationError("Experiment does not have a feature flag")
+
+        if not experiment.start_date:
+            raise ValidationError("Experiment does not have a start date")
+
+        if experiment.exposure_cohort:
+            raise ValidationError("Experiment already has an exposure cohort")
+
+        exposure_filter_data = (experiment.parameters or {}).get("custom_exposure_filter")
+        exposure_filter = None
+        if exposure_filter_data:
+            exposure_filter = Filter(data={**exposure_filter_data, "is_simplified": True}, team=experiment.team)
+
+        target_entity: int | str = "$feature_flag_called"
+        target_entity_type = "events"
+        target_filters = [
+            {
+                "key": "$feature_flag",
+                "value": [flag.key],
+                "operator": "exact",
+                "type": "event",
+            }
+        ]
```

Comment:
I understand you're creating a cohort of users exposed to the flag. But if every user requests all flags, wouldn't this simply result in a cohort of all users?

Just want to clarify what the users actually want. Do they prefer:
- a cohort of users _exposed_ to the experiment, or
- a cohort of users who _entered_ the experiment?

Maybe I'm missing something here, but to me, the latter seems more useful. So in the case of a funnel experiment, it would be a set of all users who completed the <funnel step 1> event, which has the experiment feature flag value set, and is created within the time range of the running experiment.

Reply:
cool 👍